Android: unable to use components initialProperties

### What happened?

Adding `initialProperties` to a component in the manifest causes the following crash on Android:

```
java.lang.RuntimeException: Unable to create application com.microsoft.reacttestapp.TestApp: com.squareup.moshi.JsonDataException: java.lang.ExceptionInInitializerError at $.components[0].initialProperties
  at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6739)
  at android.app.ActivityThread.access$1500(ActivityThread.java:256)
  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2090)
  at android.os.Handler.dispatchMessage(Handler.java:106)
  at android.os.Looper.loopOnce(Looper.java:201)
  at android.os.Looper.loop(Looper.java:288)
  at android.app.ActivityThread.main(ActivityThread.java:7842)
  at java.lang.reflect.Method.invoke(Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
Caused by: com.squareup.moshi.JsonDataException: java.lang.ExceptionInInitializerError at $.components[0].initialProperties
  ...
```

### Version

2.0.2

### What platforms are you seeing this issue on?

- [X] Android
- [ ] iOS
- [ ] macOS
- [ ] Windows

### Steps to Reproduce

Add `initialProperties` to the example manifest and try to build Android.
```patch
diff --git a/example/app.json b/example/app.json
index ef69fa7..aa8229f 100644
--- a/example/app.json
+++ b/example/app.json
@@ -5,7 +5,10 @@
   "components": [
     {
       "appKey": "Example",
-      "displayName": "App"
+      "displayName": "App",
+      "initialProperties": {
+        "concurrentRoot": true
+      }
     },
     {
       "appKey": "Example",

```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/android/app/src/main/java/com/microsoft/reacttestapp/TestApp.kt
+++ b/android/app/src/main/java/com/microsoft/reacttestapp/TestApp.kt
@@ -5,6 +5,7 @@ import android.app.Application
 import android.content.Context
 import com.facebook.react.PackageList
 import com.facebook.react.ReactApplication
+import com.facebook.soloader.SoLoader
 import com.microsoft.reacttestapp.manifest.ManifestProvider
 import com.microsoft.reacttestapp.react.ReactBundleNameProvider
 import com.microsoft.reacttestapp.react.TestAppReactNativeHost
@@ -24,6 +25,10 @@ class TestApp : Application(), ReactApplication {
 
     override fun onCreate() {
         super.onCreate()
+
+        // We need to initialize SoLoader early because ManifestProvider may
+        // need to use WritableNativeMap when parsing initial properties.
+        SoLoader.init(this, false)
 
         manifestProviderInternal = ManifestProvider.create(this)
         val (manifest, _) = manifestProvider.fromResources()

--- a/android/app/src/main/java/com/microsoft/reacttestapp/react/TestAppReactNativeHost.kt
+++ b/android/app/src/main/java/com/microsoft/reacttestapp/react/TestAppReactNativeHost.kt
@@ -85,7 +85,6 @@ class TestAppReactNativeHost(
         reactInstanceManager.addReactInstanceEventListener(reactInstanceListener)
 
         beforeReactNativeInit()
-        SoLoader.init(application, false)
         reactInstanceManager.createReactContextInBackground()
 
         if (BuildConfig.DEBUG) {

